plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1536,6 +1546,8 @@ private void prepareAlterTable(HdfsContext hdfsContext, HiveIdentity identity, S
             String oldTableLocation = oldTable.getStorage().getLocation();
             Path oldTablePath = new Path(oldTableLocation);
 
+            handleFilesForFailedRetryAttempts(hdfsContext, queryId, tableAndMore);
```

Comment:
Maybe something like

```
if(tableAndMore.isCheckForUnexpectedFilesInOutputDirectory()){
  removeUnexpectedFilesFromOutputDirectory(hdfsContext, queryId, directory, files)
}
```

wdyt?

Reply:
It is called in a couple of places, so I prefer one line. I can rename it to `remove...` instead `handle...` to be more explit.